### PR TITLE
contrib: Add a descriptive reference of editing primitives

### PR DESCRIPTION
**Pull request by lenormf**
Called it "trampoline" for lack of better word.

### CODE REVIEW

File: contrib/TRAMPOLINE [hunk 1]
```diff
@@ -0,0 +1,451 @@
...
+                    match those simple requirements, but it would take more
+                    work to write it than to organically apply filters on a
+                    general selection, individually.
+
+
+                    =[ SELECTION DUPLICATION
+
+        .---,       Duplicating content can be achieved using a widely
+        | y |       implemented concept: yanking and pasting. Copying the
+        `---'       current selection into the yank register (`y`) allows the
```
[the remark this answers is not in the document]
Comment (lenormf) [reply contrib/TRAMPOLINE]: That's how the documentation refers to the `"` register as, so I copied it.

File: contrib/TRAMPOLINE [hunk 2]
```diff
@@ -0,0 +1,451 @@
...
+
+                    =[ REPLACING SELECTIONS
+
+                    Text replacement is a two step process in Kakoune, which
+                    involves selecting text to be replaced, and then erasing it
+        .---,       to insert the replacement text. After selections have been
+        | c |       made, you can simply hit the deletion primitive (`d`), then
+        `---'       either enter insert mode to write down the replacement text
+                    (`i`), or stay in command mode to paste the replacement
+                    text stored in the yank register. As deleting and entering
```
[the remark this answers is not in the document]
Comment (lenormf) [reply contrib/TRAMPOLINE]: I assumed new users would remove some text, go yank the replacement text somewhere in the buffer, then go back to the deletion to paste what they just copied. I'll just remove the second part of this sentence, as I don't want to talk about `<a-d>` at all.

File: contrib/TRAMPOLINE [hunk 3]
```diff
@@ -0,0 +1,451 @@
...
+           | O |    for each of them is achieved by passing the amount of lines
+           `---'    as a count respectively to the `o` and `O` primitives.
+
+
+                    =[ REGISTERS
+
+                    Similarly to counts, registers influence the behavior of
+   .---,            certain primitives. They are storage structures identified
+   | " |_.          by a single character, and are populated by primitives as a
+   `---'  `.---,    result of a side effect. Although primitives populate a
```
[the remark this answers is not in the document]
Comment (lenormf) [reply contrib/TRAMPOLINE]: Some primitives' side effect result in registers being changed.

Using "primitive" for keys is less confusing, as I assume most users see commands as whatever name they type in the prompt after hitting `:`.

File: contrib/TRAMPOLINE (hunk 1 above)
[the remark this answers is not in the document]
Comment (mawww) [reply contrib/TRAMPOLINE]: Well, I'd rather fix the `"` register's documentation then.

File: contrib/TRAMPOLINE (hunk 3 above)
[the remark this answers is not in the document]
Comment (mawww) [reply contrib/TRAMPOLINE]: But `:doc commands` actually calls the `:` commands `primitives`... The confusion is getting stronger now.

File: contrib/TRAMPOLINE (hunk 1 above)
[the remark this answers is not in the document]
Comment (lenormf) [reply contrib/TRAMPOLINE]: I replaced "yank register" with "copy register", including in the documentation, is that better?

File: contrib/TRAMPOLINE (hunk 3 above)
[the remark this answers is not in the document]
Comment (lenormf) [reply contrib/TRAMPOLINE]: I renamed the "Primitives" section into "Builtins" in the documentation.

**Pull request merged**